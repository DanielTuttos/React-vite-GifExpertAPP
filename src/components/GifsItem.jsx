import PropTypes from 'prop-types';

export const GifsItem = ({ title, url }) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifsItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}