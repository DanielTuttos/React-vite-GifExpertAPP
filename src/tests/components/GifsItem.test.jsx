import { render } from "@testing-library/react";
import { GifsItem } from "../../components/";

describe('Tests in <GifsItem />', () => {

    const title = "Saitama";
    const url = "https://one-puch.com/saitama.jpg";

    test('must have match with snapshot', () => {
        const { container } = render(<GifsItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

})