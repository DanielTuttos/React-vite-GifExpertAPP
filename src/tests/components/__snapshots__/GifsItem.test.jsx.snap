// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifsItem /> must have match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-puch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
